lib/grape/validations/validators/values.rb

```diff
@@ -43,7 +43,11 @@ def validate_param!(attr_name, params)
       def check_values(param_array)
         values = @values.is_a?(Proc) && @values.arity.zero? ? @values.call : @values
```

Comment:
Why aren't we also rescuing this `@values.call` here in that case? (and would need a spec)

Reply:
I think that an error raised from an arity-zero proc is a different and more serious condition.  The output there is expected to be an array of valid values (or invalid values in the case of except_values) and a raised error means that the validator is completely broken. Yes, we _could_ return an empty array and warn in that case, but that doesn't feel like the right thing to do. In the values case it wouldn't ever accept anything; and, worse, in the except_values case it would always accept everything.

The primary reason I want to rescue in the arity-one case is because of the likelihood that unexpected input values (non-numeric string or nil in the example above) will raise errors in simple validation code that doesn't do thorough type-safety checks.  But in the arity-zero case, there is no input value so that's not a consideration.